spec/services/invoices/retry_service_spec.rb

```diff
@@ -0,0 +1,276 @@
+        :subscription,
+        plan:,
+        subscription_at: started_at,
+        started_at:,
+        created_at: started_at
+      )
+    end
+
+    let(:timestamp) { Time.zone.now - 1.year }
+    let(:started_at) { Time.zone.now - 2.years }
+    let(:plan) { create(:plan, organization:, interval: 'monthly') }
+    let(:billable_metric) { create(:billable_metric, aggregation_type: 'count_agg') }
+    let(:charge) { create(:standard_charge, plan: subscription.plan, charge_model: 'standard', billable_metric:) }
+
+    let(:fee_subscription) do
+      create(
+        :fee,
+        invoice:,
+        subscription:,
+        fee_type: :subscription,
+        amount_cents: 2_000
+      )
+    end
+    let(:fee_charge) do
+      create(
+        :fee,
+        invoice:,
+        charge:,
+        fee_type: :charge,
+        total_aggregated_units: 100,
+        amount_cents: 1_000
+      )
+    end
+
+    let(:integration) { create(:anrok_integration, organization:) }
+    let(:integration_customer) { create(:anrok_customer, integration:, customer:) }
+    let(:response) { instance_double(Net::HTTPOK) }
+    let(:lago_client) { instance_double(LagoHttpClient::Client) }
+    let(:endpoint) { 'https://api.nango.dev/v1/anrok/finalized_invoices' }
+    let(:body) do
+      path = Rails.root.join('spec/fixtures/integration_aggregator/taxes/invoices/success_response_multiple_fees.json')
+      json = File.read(path)
+
+      # setting item_id based on the test example
+      response = JSON.parse(json)
+      response['succeededInvoices'].first['fees'].first['item_id'] = subscription.id
+      response['succeededInvoices'].first['fees'].last['item_id'] = billable_metric.id
+
+      response.to_json
+    end
+    let(:integration_collection_mapping) do
+      create(
+        :netsuite_collection_mapping,
+        integration:,
+        mapping_type: :fallback_item,
+        settings: {external_id: '1', external_account_code: '11', external_name: ''}
+      )
+    end
+
+    before do
+      integration_collection_mapping
+      fee_subscription
+      fee_charge
+
+      allow(SegmentTrackJob).to receive(:perform_later)
+      allow(Invoices::Payments::StripeCreateJob).to receive(:perform_later).and_call_original
+      allow(Invoices::Payments::GocardlessCreateJob).to receive(:perform_later).and_call_original
+
+      integration_customer
+
+      allow(LagoHttpClient::Client).to receive(:new).with(endpoint).and_return(lago_client)
+      allow(lago_client).to receive(:post_with_response).and_return(response)
+      allow(response).to receive(:body).and_return(body)
+    end
+
+    context 'when invoice does not exist' do
+      it 'returns an error' do
+        result = described_class.new(invoice: nil).call
+
+        expect(result).not_to be_success
+        expect(result.error.error_code).to eq('invoice_not_found')
+      end
+    end
+
+    context 'when invoice is in draft status' do
+      before do
+        invoice.update(status: :draft)
```

Comment:
We can test all statuses that should return an error (all except `failed`):

```suggestion
        invoice.update(status: %i[draft finalized voided generating].sample)
```